diff --git a/meson.build b/meson.build
index 0b8a405..8abfdf8 100644
--- a/meson.build
+++ b/meson.build
@@ -28,7 +28,7 @@ endif

 curl_dep = dependency('libcurl')

-if build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); }; void main() {}')
+if build_machine.system() == 'darwin' or build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); }; void main() {}')
     argplib = cc.find_library('argp')
 else
     argplib = dependency('', required : false)
diff --git a/src/lib/index/index_read.c b/src/lib/index/index_read.c
index c6ee98b..875b569 100644
--- a/src/lib/index/index_read.c
+++ b/src/lib/index/index_read.c
@@ -30,6 +30,8 @@
 #include <string.h>
 #ifdef FREEBSD
 #include <sys/endian.h>
+#elif __APPLE__
+#include <machine/endian.h>
 #else
 #include <endian.h>
 #endif